cumulus_library/studies/core/builder_encounter_coding.py

```diff
@@ -0,0 +1,123 @@
+""" Module for generating condition codeableConcept table"""
+
+from typing import Dict, List
+
+from cumulus_library.base_table_builder import BaseTableBuilder
+from cumulus_library.helper import get_progress_bar, query_console_output
+from cumulus_library.template_sql.templates import (
+    CodeableConceptConfig,
+    get_codeable_concept_denormalize_query,
+    get_is_table_not_empty_query,
+)
+
+
+class EncounterCodingBuilder(BaseTableBuilder):
+    display_text = "Creating encounter type code table..."
+
+    def _check_data_in_fields(self, code_sources: List[Dict], cursor, progress, task):
+        """checks if CodableConcept fields actually have data available"""
+
+        # TODO: consider moving to a utility library if we have another case like
+        # this one
+        cols_with_data = []
+        for code_source in code_sources:
```

Comment:
This method contains some hairy nested logic. A newcomer might appreciate (a) a dollop of "what is happening" comments maybe for the separate execution steps and (b) a dollop of "why is this happening" maybe in the docstring, for like, why this method is relevant/useful in the broader context.

Reply:
i mean, :i: hate it, so lemme try to be more descriptive here.